src/argilla/client/sdk/datasets/models.py

```diff
@@ -36,7 +36,7 @@ def _missing_(cls, value):
 class BaseDatasetModel(BaseModel):
     tags: Dict[str, str] = Field(default_factory=dict)
     metadata: Dict[str, Any] = Field(default_factory=dict)
-    name: str = Field(regex="^(?!-|_)[a-z0-9-_]+$")
+    name: str = Field(regex="^(?!-|_)[a-zA-Z0-9-_]+$")
```

Comment:
For old datasets this is not a valid name, we should revert this change here

Reply:
You're right! I'll rever the `regex` validation update there!